src/Cache/Dragon/Dragon.js

```diff
@@ -5,61 +5,81 @@ import CopyData from '../../Sim/SimulationData/CopyData';
 import { SimDoSims, SimObjects } from '../../Sim/VariablesAndData';
 import { CacheCostDragonUpgrade, CacheLastDragonLevel } from '../VariablesAndData'; // eslint-disable-line no-unused-vars
 
+
+// Calculate the cost to buy up the given number of previously bought levels of
+// the building. This function does not set up the simulation data
+// automatically; this must be handled by the caller.
+function CalculateRebuyCostOfPreviousLevels(building, levels) {
+  let cost = 0;
+  for (let i = 0; i < levels; i++) {
+    const simTarget = SimObjects[building.name];
+    let price =
+      simTarget.basePrice *
+      Game.priceIncrease **
+        Math.max(0, simTarget.amount - 1 - simTarget.free);
+    price = Game.modifyBuildingPrice(simTarget, price);
+    price = Math.ceil(price);
+    cost += price;
+    simTarget.amount -= 1;
+  }
+  return cost;
+}
+
+function GetDragonCostTooltipText() {
+  if (Game.dragonLevel < 5) {
+    // Early levels cost cookies, no extra calculation needed.
+  } else if (Game.dragonLevel < 23) {
+    // Dragon levels cost X number of a specific building
+    const target = Game.ObjectsById[Game.dragonLevel-5];
+    if (!target) {
+      return 'Cost to rebuy: unknown'
+    }
+
+    const match = Game.dragonLevels[Game.dragonLevel].buy.toString()
+      .match(/sacrifice\((.*?)\)/);
+    if (!match) {
+      return 'Cost to rebuy: unknown'
+    }
+    const amount = parseInt(match[1], 10);
+    if (target.amount < amount) {
+      return 'Not enough buildings to sacrifice';
+    }
+
+    CopyData();
+    const cost = CalculateRebuyCostOfPreviousLevels(target, amount);
+    return `Cost to rebuy: ${Beautify(cost)}`;
+  } else if (Game.dragonLevel < 25) {
+    // Dragon levels cost X of every building
+    const match = Game.dragonLevels[Game.dragonLevel].buy.toString()
+      .match(/sacrifice\((.*?)\)/);
+    if (!match) {
+      return 'Cost to rebuy: unknown'
+    }
+    const amount = parseInt(match[1], 10);
+
+    let cost = 0;
+    CopyData();
+    for (const target of Object.values(Game.Objects)) {  // eslint-disable-line no-restricted-syntax
```

Comment:
Can't we do this without disabling the `eslint` message?

Reply:
Sure. I wanted to avoid calling `hasOwnProperty` to make the code cleaner. We can't use `forEach` because we cannot do early return with it. I'll rewrite this into a `for .. in` loop.

I wonder why `for .. of` loops are restricted in 2021.

Reply:
Ah, nevermind. Aibnb config forbids the use of all `for` loops. This is an *interesting* choice. and it has been a point of great debate over at their repo: https://github.com/airbnb/javascript/issues/1271

I can do the `target.amount < amount` checks before the main calculation loop.